sourcegraph/sourcegraph
# Add feature flag to disable proactive aggregation insights

Closes https://github.com/sourcegraph/sourcegraph/issues/40427
Closes https://github.com/sourcegraph/sourcegraph/issues/40683

## Test plan

- Unit test should pass
- Feature flag `search-aggregation-filters` need to be enabled
  - Go to `site-admin/feature-flags` create a boolean feature flag called `search-aggregation-filters` set to true
- Search for anything. i.e. "hello" Confirm aggregation card shows in the side panel
- Enable `disable-proactive-insight-aggregation`
  - Go to `site-admin/feature-flags` create a boolean feature flag called `disable-proactive-insight-aggregation` set to true
- Run any search
- Confirm only buttons show in the side bar and explanatory text show

![image](https://user-images.githubusercontent.com/1855233/186290127-331dafc3-3123-4949-9498-416d9a560521.png)

- Select a grouping mode by clicking one of the buttons
- Confirm chart displays

## App preview:

- [Web](https://sg-web-insights-disable-proactive.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-ffqjudufiw.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
